source/util/AMTrapezoidVelocityProfile.h

```diff
@@ -0,0 +1,82 @@
+#ifndef AMTRAPEZOIDVELOCITYPROFILE_H
+#define AMTRAPEZOIDVELOCITYPROFILE_H
+
+/*!
+ * \brief A class which represents the velocity profile of a some motion which
+ * follows a trapezoidal structure ie. it begins at some base velocity, accelerating
+ * at the initialAcceleration towards some target velocity; when it reached the
+ * point at which it must decelerate to return to the base velocity by the end
+ * of the motion changing velocity at the rate defined by returnAcceleration it
+ * will begin its deceleration.
+ * NOTE 1: If the time of the motion is not sufficient to accelerate up to the
+ * target velocity, the profile will resemble a triangle, rather than a trapezoid.
+ * NOTE 2: The velocity profile might describe a motion in which the target velocity
```

Comment:
Where the target velocity is less than the base? Well I wanted to make it fairly general, but I could also imagine the case where you're at position 20 and you would like to move to position 10 in 5 seconds; in which case your base velocity will be 0 and your target -2.